### PR TITLE
qt: Add "directory" option to FileDialog & Dialog

### LINKED ISSUE #874: File.openDialog doesn't allow to start browsing from other directory than $HOME
it would be great if it was possible to pass on a further argument to .openDialog in order to determine a different starting directory for browsing than $HOME, which is currently hard-coded in QcFileDialog.cpp.

### PR DESCRIPTION
**Pull request by snappizz**
fixes #874.

### CODE REVIEW

File: QtCollider/widgets/QcFileDialog.h [hunk 1]
```diff
@@ -34,7 +34,8 @@ class QcFileDialog : public QObject
 public:
 
   Q_INVOKABLE QcFileDialog( int fileMode = QFileDialog::ExistingFile,
-                            int acceptMode = QFileDialog::AcceptOpen );
+                            int acceptMode = QFileDialog::AcceptOpen,
+                            const QString& directory = "" );
```
Comment (snappizz) [inline QtCollider/widgets/QcFileDialog.h]: i am not sure how to make this default to `QDir::home()`

**Comment by brianlheim** [comment]
:/ I was just getting ready to push a branch that does the same thing.

File: QtCollider/widgets/QcFileDialog.h (hunk 1 above)
Comment (brianlheim) [reply QtCollider/widgets/QcFileDialog.h]: `= QDir::home().path()`

**Comment by brianlheim** [comment]
@snappizz - I have some other documentation changes that add examples and rewrite most of the doc for FileDialog. Mind if I push them here?

**Comment by brianlheim** [comment]
Also, wonder what you would think of changing `directory` to `path`. A little more succinct, but either works.

File: SCClassLibrary/Common/GUI/Base/QDialog.sc [hunk 2]
```diff
@@ -40,13 +43,13 @@ FileDialog : QObject {
 
 Dialog {
 
-	*openPanel { arg okFunc, cancelFunc, multipleSelection=false;
+	*openPanel { arg okFunc, cancelFunc, multipleSelection=false, directory;
 		var fileMode;
 		if( multipleSelection ) { fileMode = 3 } { fileMode = 1 };
-		^FileDialog.new( okFunc, cancelFunc, fileMode, 0, stripResult:multipleSelection.not );
+		^FileDialog.new( okFunc, cancelFunc, fileMode, 0, stripResult:multipleSelection.not, directory: directory );
```
Comment (brianlheim) [inline SCClassLibrary/Common/GUI/Base/QDialog.sc]: the args are already in correct order, so you could remove `stripResult:` and `directory:` here for simplicity

Comment (telephon) [reply SCClassLibrary/Common/GUI/Base/QDialog.sc]: … also for speed.

**Pull request closed without merging**